src/components/private/onsite-payment.tsx

```
@@ -0,0 +1,128 @@
+    const order = orders.find((o) => o.id.toLowerCase() === orderId.toLowerCase());
+    if (order) {
+      setSelectedOrder(order);
+    } else {
+      setSelectedOrder(null);
+    }
+  };
+
+  const handleConfirm = () => {
+    if (selectedOrder && amount && paymentOption) {
+      const paymentAmount = Number.parseFloat(amount);
+      if (!isNaN(paymentAmount) && paymentAmount > 0 && paymentAmount <= selectedOrder.balance) {
```

Biome (1.9.4)

[error] isNaN is unsafe. It attempts a type coercion. Use Number.isNaN instead.

See the MDN documentation for more details.
Unsafe fix: Use Number.isNaN instead.


(lint/suspicious/noGlobalIsNan)